ostd/src/cpu/mod.rs

```diff
@@ -169,26 +29,35 @@ unsafe impl PinCurrentCpu for dyn InAtomicMode + '_ {}
 pub(crate) unsafe fn init_on_bsp() {
     let num_cpus = crate::arch::boot::smp::count_processors().unwrap_or(1);
 
-    // SAFETY: The safety is upheld by the caller and
-    // the correctness of the `get_num_processors` method.
+    // SAFETY: The safety conditions of `init_on_bsp` correspond to
+    // whose of `local::copy_bsp_for_ap`.
     unsafe {
         local::copy_bsp_for_ap(num_cpus as usize);
+    }
 
-        set_this_cpu_id(0);
-
-        // Note that `init_num_cpus` should be called after `copy_bsp_for_ap`.
-        // This helps to build the safety reasoning in `CpuLocal::get_on_cpu`.
-        // See its implementation for details.
-        init_num_cpus(num_cpus);
+    // SAFETY:
+    // 1. The #1 and #2 safety conditions of `init_on_bsp` correspond to
+    // the first two of `id::init_on_bsp`.
+    // 2. The previous step of `copy_bsp_for_ap` ensures that
+    // the CPU-local storage is safe to use.
```

Comment:
In doc comments, rustfmt will add some space like this:
```rust
    // 1. The #1 and #2 safety conditions of `init_on_bsp` correspond to
    //    the first two of `id::init_on_bsp`.
    // 2. The previous step of `copy_bsp_for_ap` ensures that
    //    the CPU-local storage is safe to use.
```

This is not a doc comment, so we have to do this manually. I think it looks better this way.

However, as I said above, I don't think the second safety comment makes much sense, so I think it should be removed.